[FEAT] 수강생 관리 과제 조회 구현

cuteJJong opened the pull request:
## 관련 이슈
<!-- 관련된 이슈 번호를 적어주세요. -->
<!-- ex) Closes #이슈번호 -->
- Closes #104 

## 변경 사항
<!-- 이 PR에서 변경된 주요 내용을 설명해주세요.
     변경사항이 여러 개일 경우, 주요 내용과 하위 항목을 구분하여 작성해주세요. -->
<!-- ex) 1. 000 엔티티 구현 -->
<!--        - 000 메서드 구현 -->
1. 수강생 관리 과제 조회 구현

## 스크린샷
<!-- (선택) 주요 변경사항(기능 구현 화면, 테스트 결과 등)에 대한 이미지를 삽입해주세요. -->
<!-- 변경사항에 따라 아래 표 템플릿을 활용해주세요 -->

|기능|스크린샷|
|---|---|
|수강생관리 과제 조회|<img width="508" alt="image" src="https://github.com/user-attachments/assets/e6a20fb5-5429-4e9e-b6bf-5ef15b264d9f">|

## 체크리스트
<!-- PR을 제출하기 전에 확인해야 할 사항들입니다. -->
- [x] 코드 컨벤션을 준수하였습니까?
- [x] 모든 테스트를 통과하였습니까?
- [x] 관련 문서를 업데이트하였습니까?

cuteJJong committed:
FEAT: 과제 조회 기능 구현

cuteJJong committed:
FEAT: 강의에 대한 과제 전체 조회 API 구현

cuteJJong committed:
Merge branch 'main' into feature/104-frontend-student-management

cuteJJong committed:
REFACTOR: 필요없는 API 삭제

cuteJJong committed:
"FEAT: useNavigate 추가로 내비게이션 기능 도입

cuteJJong committed:
FEAT: 학생별 과제 현황 조회 구현

cuteJJong committed:
Merge branch 'main' into feature/104-frontend-student-management

liimjiin reviewed (approved):
진행 상황까지 구현이 잘 되어있네요. 고생하셨습니다!

sseung519 reviewed (approved):
수강생 관리 > 과제 제출 현황까지 잘 구현되었군요
고생하셨습니다.

cuteJJong committed:
FEAT: 시험 채점 및 결과 조회 라우팅 추가

cuteJJong committed:
FEAT: 시험 채점 및 결과 조회 라우팅 추가

cuteJJong committed:
REFACTOR: 시험 관련 권한 구조 정리

- FEATURE_PERMISSIONS에 중복된 권한 제거
  - GRADE_EXAM과 EXAM_GRADING 통합
  - VIEW_OWN_RESULT와 EXAM_VIEW_RESULT 통합
- 권한 이름을 더 명확하게 수

cuteJJong committed:
FEAT: 시험 채점 및 결과 조회 관련 API 추가

cuteJJong committed:
REFACTOR: 시험 목록 페이지 UI/UX 개선 및 역할 기반 기능 분리

cuteJJong committed:
FEAT: 시험 답안 채점 폼 컴포넌트 구현

cuteJJong committed:
FEAT: 시험 채점 폼 페이지 레이아웃 구현

cuteJJong committed:
FEAT: 시험 채점 목록 및 상세 조회 페이지 구현

cuteJJong committed:
FEAT: 시험 채점 목록 페이지 권한 처리 구현

cuteJJong committed:
FEAT: 시험 결과 조회 페이지 구현

cuteJJong committed:
FEAT: 역할 기반 시험 결과 페이지 구현

cuteJJong committed:
REFACTOR: 시험 응시 페이지 개선 및 안정성 강화

cuteJJong committed:
FEAT: 학생용 시험 결과 조회 페이지 구현

cuteJJong committed:
FEAT: 학생용 시험 결과 페이지 레이아웃 구현

cuteJJong committed:
MERGE: 메인 브랜치와 작업 브랜치 병합 및 충돌 해결

cuteJJong committed:
REFACTOR: 시험 관련 로직 수정

cuteJJong committed:
Merge branch 'main' into feature/104-frontend-student-management
